tests/foreman/virtwho/test_cli_virtwho.py

```diff
@@ -14,82 +14,327 @@
-
-    :id: 562c5054-49ba-4c89-9db7-73659c7ac9ef
-
-    :steps:
-        1. Create config by hammer virt-who-config create
-        2. Check the config id by hammer virt-who-config list
-        3. Deploy the config by hammer virt-who-config deploy
-        4. Update VIRTWHO_INTERVAL to differernt value
-           by hammer virt-who-config update
-        5. Deploy the config again by hammer virt-who-config deploy
-
-    :expectedresults:
-        1. Config can be created and deployed
-        2. Config VIRTWHO_INTERVAL be updated in /etc/sysconfig/virt-who
-
-    :CaseAutomation: notautomated
-    """
+from robottelo.decorators import tier1
+from robottelo.test import CLITestCase
+from robottelo.config import settings
+from robottelo.cli.virt_who_config import VirtWhoConfig
+from robottelo.cli.subscription import Subscription
+from robottelo.cli.host import Host
+from robottelo.cli.user import User
+from robottelo.constants import DEFAULT_ORG
+from fauxfactory import gen_string
+from .utils import (
+    deploy_configure_by_command,
+    deploy_configure_by_script,
+    get_configure_file,
+    get_configure_option,
+    get_configure_command,
+    VIRTWHO_SYSCONFIG,
+)
+
+
+class VirtWhoConfigTestCase(CLITestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super(VirtWhoConfigTestCase, cls).setUpClass()
+        cls.satellite_url = settings.server.hostname
+        cls.hypervisor_type = settings.virtwho.hypervisor_type
+        cls.hypervisor_server = settings.virtwho.hypervisor_server
+        cls.hypervisor_username = settings.virtwho.hypervisor_username
+        cls.hypervisor_password = settings.virtwho.hypervisor_password
+        cls.hypervisor_config_file = settings.virtwho.hypervisor_config_file
+        cls.vdc_physical = settings.virtwho.sku_vdc_physical
+        cls.vdc_virtual = settings.virtwho.sku_vdc_virtual
+
+    def _make_virtwho_configure(self):
+        args = {
+            u'debug': 1,
+            u'interval': '60',
+            u'hypervisor-id': 'hostname',
+            u'hypervisor-type': self.hypervisor_type,
+            u'hypervisor-server': self.hypervisor_server,
+            u'organization-id': 1,
+            u'filtering-mode': 'none',
+            u'satellite-url': self.satellite_url,
+        }
+        if self.hypervisor_type == 'libvirt':
+            args[u'hypervisor-username'] = self.hypervisor_username
+        elif self.hypervisor_type == 'kubevirt':
+            args[u'kubeconfig'] = self.hypervisor_config_file
+        else:
+            args['hypervisor-username'] = self.hypervisor_username
+            args['hypervisor-password'] = self.hypervisor_password
+        return args
+
+    @tier1
+    def test_positive_deploy_configure_by_id(self):
+        """ Verify " hammer virt-who-config deploy"
+
+        :id: 19ffe76e-7e3d-48c7-b846-10a83afe0f3e
+
+        :expectedresults: Config can be created and deployed
+
+        :CaseLevel: Integration
+
+        :CaseImportance: High
```

Comment:
as per test it looks `CaseImportance`: `Critical` , What do you think ?

Reply:
In my understand about from the 'Test metadata documentation', if the product release will be blocked or pending by the failed cases, we should set it to 'Critical'.  
Virtwho Configure is not a mandatory plugin for satellite, without it, the customers still can create the configure by manual with the virt-who package.

Reply:
make sense :+1: